src/deployment.ts

```diff
@@ -161,6 +174,7 @@ export class Deployment extends workload.Workload implements IScalable {
 
     this.replicas = props.replicas;
     this.strategy = props.strategy ?? DeploymentStrategy.rollingUpdate();
+    this.revisionHistoryLimit = props.revisionHistoryLimit;
```

Comment:
We normally apply kubernetes defaults in order to surface the value in the manifest. 
 
```suggestion
    this.revisionHistoryLimit = props.revisionHistoryLimit ?? 10;
```

Also this needs a unit test which passes a custom value and asserts it indeed propagates to the manifest.